dev-util/gn/patches/gn-01463.patchset

```diff
@@ -0,0 +1,440 @@
+From 3c1b150235ce61ef80463dcf92da4d05dae45c58 Mon Sep 17 00:00:00 2001
+From: Calvin Hill <calvin@hakobaito.co.uk>
+Date: Wed, 26 Sep 2018 20:48:23 +0100
+Subject: gn: Haiku support
+
+
+diff --git a/base/files/file.h b/base/files/file.h
+index 61239ad..bea5094 100644
+--- a/base/files/file.h
++++ b/base/files/file.h
+@@ -22,10 +22,10 @@
+ 
+ namespace base {
+ 
+-#if defined(OS_BSD) || defined(OS_MACOSX) || defined(OS_NACL) || \
++#if defined(OS_BSD) || defined(OS_MACOSX) || defined(OS_NACL) || defined(OS_HAIKU) || \
+     defined(OS_ANDROID) && __ANDROID_API__ < 21
+ typedef struct stat stat_wrapper_t;
+-#elif defined(OS_POSIX) || defined(OS_FUCHSIA)
++#elif defined(OS_POSIX) || defined(OS_FUCHSIA) && !defined(OS_HAIKU)
+ typedef struct stat64 stat_wrapper_t;
+ #endif
+ 
+diff --git a/base/files/file_path.cc b/base/files/file_path.cc
+index a66c8cc..c9996b2 100644
+--- a/base/files/file_path.cc
++++ b/base/files/file_path.cc
+@@ -415,7 +415,7 @@ FilePath FilePath::InsertBeforeExtensionASCII(StringPiece suffix) const {
+   DCHECK(IsStringASCII(suffix));
+ #if defined(OS_WIN)
+   return InsertBeforeExtension(ASCIIToUTF16(suffix));
+-#elif defined(OS_POSIX) || defined(OS_FUCHSIA)
++#elif defined(OS_POSIX) || defined(OS_FUCHSIA) || defined(OS_HAIKU)
```

Comment:
Since we do follow the POSIX spec, shouldn't we be classified as `OS_POSIX`?

Reply:
@alaviss I've already added Haiku as a POSIX platform in the [patchset](https://github.com/haikuports/haikuports/pull/3306/files#diff-c150c993afd02ddd4c9be7a05a33fd56R355), but yes the extra OS_HAIKU macro is pretty much superfluous. I'll clean it up to use OS_POSIX instead.